src/Microsoft.AspNet.Mvc.ModelBinding/Binders/CompositeModelBinder.cs

```diff
@@ -34,14 +33,17 @@ public CompositeModelBinder([NotNull] IEnumerable<IModelBinder> modelBinders)
 
         public virtual async Task<ModelBindingResult> BindModelAsync([NotNull] ModelBindingContext bindingContext)
         {
-            var newBindingContext = CreateNewBindingContext(bindingContext,
-                                                            bindingContext.ModelName);
-
+            var newBindingContext = CreateNewBindingContext(bindingContext, bindingContext.ModelName);
             var modelBindingResult = await TryBind(newBindingContext);
-            if (modelBindingResult == null && !string.IsNullOrEmpty(bindingContext.ModelName)
-                && bindingContext.FallbackToEmptyPrefix)
+
+            if (modelBindingResult == null &&
+                bindingContext.FallbackToEmptyPrefix &&
```

Comment:
mostly. I was mainly re-wrapping the condition because it looked strange but chose this order because it's `FallbackToEmptyPrefix` that makes `IsNullOrEmpty` relevant.